### ISSUE TITLE
Tabs inserted instead of spaces in makefiles

### ISSUE THREAD

**Issue by kowalski5233** [microsoft/vscode#144375]
Issue Type: <b>Bug</b>

I have an issue where, even though Editor: Insert Spaces is true and Editor: Detect Indentation is false, it keeps inserting actual tabs when I press Tab in a makefile (.mk).  I have even added the "editor.insertSpaces": true, option for the overwriting settings for makefiles, restarted the IDE, and still it inserts tabs instead of spaces.  The "editor.detectIndentation": false, was already in the settings for makefiles.

Version: 1.64.2 (user setup)
Commit: f80445acd5a3dadef24aa209168452a3d97cc326
Date: 2022-02-09T22:02:28.252Z
Electron: 13.5.2
Chromium: 91.0.4472.164
Node.js: 14.16.0
V8: 9.1.269.39-electron.0
OS: Windows_NT x64 10.0.22000

VS Code version: Code 1.64.2 (f80445acd5a3dadef24aa209168452a3d97cc326, 2022-02-09T22:02:28.252Z)
OS version: Windows_NT x64 10.0.22000
Restricted Mode: No

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-8665U CPU @ 1.90GHz (8 x 2112)|
|GPU Status|2d_canvas: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: enabled<br>opengl: enabled_on<br>rasterization: enabled<br>skia_renderer: enabled_on<br>video_decode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|15.75GB (3.70GB free)|
|Process Argv|--crash-reporter-id a1a7a1da-a69b-47dd-b768-38c15a4c3e53|
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (18)</summary>

Extension|Author (truncated)|Version
---|---|---
doxdocgen|csc|1.4.0
vscode-pull-request-github|Git|0.36.2
better-cpp-syntax|jef|1.15.13
python|ms-|2022.2.1924087327
vscode-pylance|ms-|2022.3.0
jupyter|ms-|2022.1.1301854968
jupyter-keymap|ms-|1.0.0
jupyter-renderers|ms-|1.0.6
remote-containers|ms-|0.224.2
remote-ssh|ms-|0.76.1
remote-ssh-edit|ms-|0.76.1
remote-wsl|ms-|0.64.2
cmake-tools|ms-|1.9.2
cpptools|ms-|1.9.2
cpptools-extension-pack|ms-|1.1.0
vsliveshare|ms-|1.0.5418
vsliveshare-audio|ms-|0.1.91
cmake|twx|0.0.17

(1 theme extensions excluded)

</details><details>
<summary>A/B Experiments</summary>

```
vsliv368:30146709
vsreu685:30147344
python383:30185418
vspor879:30202332
vspor708:30202333
vspor363:30204092
pythonvspyl392:30443607
pythontb:30283811
pythonvspyt551cf:30345471
pythonptprofiler:30281270
vshan820:30294714
vstes263cf:30335440
vscoreces:30445986
pythondataviewer:30285071
vscod805:30301674
pythonvspyt200:30340761
binariesv615:30325510
bridge0708:30335490
bridge0723:30353136
vsaa593cf:30376535
vsc1dst:30438360
pythonvs932:30410667
wslgetstarted:30433507
vsclayoutctrc:30437038
vsrem710:30416614
vscop841:30438915
dsvsc008:30440022
pythonvspyt640:30442238
vsbas813:30436447
vscscmwlcmt:30438805
helix:30440343
vsnot107:30443614

```

</details>

**Comment by alexdima** [comment]
@kowalski5233 Extensions can also tweak the current text editor's indentation settings. Does this problem reproduce when disabling extensions `Code.exe --disable-extensions` ?

**Comment by JDdeVilliers** [comment]
Hi @alexdima , thanks for the reply.  This still appears to be the case, even with extensions turned off.  However, I wonder if it's not overwritten by VS entirely, because makefiles uses spaces for indentations and tabs for recipes, something I didn't really think of at the time.  So perhaps it overrides any preference so that tab is always tab, and space is always space, for makefiles?

**Comment by alexdima** [comment]
@JDdeVilliers You're right, it looks like we ship with a language override for `editor.insertSpaces` for `makefile`. This can be seen when running `F1 > Open Default Settings (JSON)`. Unfortunately default language specific settings trump language agnostic user settings, so to change this setting for makefile, you will need to use a language specific setting: i.e.:

```
	"[makefile]":  {
		"editor.insertSpaces": true
	},
```
